src/memray/commands/transform.py

```diff
@@ -0,0 +1,94 @@
+import argparse
+import shutil
+import sys
+from typing import Any
+from typing import cast
+
+from rich import print as pprint
+
+from memray._errors import MemrayCommandError
+
+from ..reporters.transform import TransformReporter
+from .common import HighWatermarkCommand
+from .common import ReporterFactory
+
+
+class TransformCommand(HighWatermarkCommand):
+    """Generate reports files in different formats"""
+
+    def __init__(self) -> None:
+        super().__init__(
+            reporter_factory=cast(ReporterFactory, TransformReporter),
+            reporter_name="transform",
+        )
+
+    def prepare_parser(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "format",
+            help="Format to use for the report",
+        )
+        parser.add_argument(
+            "-o",
+            "--output",
+            help="Output file name",
+            default=None,
+        )
+        parser.add_argument(
+            "-f",
+            "--force",
+            help="If the output file already exists, overwrite it",
+            action="store_true",
+            default=False,
+        )
+        parser.add_argument(
+            "--leaks",
+            help="Show memory leaks, instead of peak memory usage",
+            action="store_true",
+            dest="show_memory_leaks",
+            default=False,
+        )
+        parser.add_argument("results", help="Results of the tracker run")
+
+    def run(
+        self, args: argparse.Namespace, parser: argparse.ArgumentParser, **kwargs: Any
+    ) -> None:
+        the_format = args.format.lower()
+        suffix = TransformReporter.SUFFIX_MAP.get(the_format)
+        if not suffix:
+            raise MemrayCommandError(
+                f"Format not supported: {args.format}", exit_code=1
+            )
+
+        self.suffix = suffix
+        super().run(args, parser, format=the_format)
+
+        post_run_callable = getattr(self, f"post_run_{the_format}", None)
+        if post_run_callable:
+            post_run_callable()
+
+    def post_run_gprof2dot(self) -> None:
+        assert self.output_file is not None
+        try:
+            import gprof2dot  # noqa
+        except ImportError:
+            pprint(
+                ":exclamation: [red]gprof2dot doesn't seem to be installed. "
+                "Please install it to be able to process the transform.[/red]"
+                "\n:point_right: Check out https://github.com/jrfonseca/gprof2dot for "
+                "installation instructions"
+            )
```

Comment:
This assumes that `gprof2dot` is gonna be pip installed into the same virtualenv as Memray is running in, but it can also be installed from a distro package manager, in which case this results in a false positive. Maybe we should also try running `shutil.which("gprof2dot")`? Then if `which("gprof2dot")` works, we'd use just `gprof2dot -f json` instead of `python -m gprof2dot -f json` in the the message below.

This isn't just theoretical, when I tried running this before reading the code, I got this warning message, and then I ran `apt-get install gprof2dot`, and I was confused when that didn't make the warning go away.